packages/snaps-rpc-methods/src/permitted/revokePermissions.ts

```diff
@@ -0,0 +1,117 @@
+
+import type { MethodHooksObject } from '../utils';
+
+/**
+ * The revokePermissions hook parameters used by the `snap_revokePermissions` method.
+ *
+ */
+export type RevokePermissionsHookParams = (keyof InitialPermissions)[];
+
+export type RevokePermissionsHooks = {
+  getOriginSnap: () => Snap | undefined;
+  /**
+   * @returns Null.
+   */
+  revokePermissions: (
+    dynamicPermissions: RevokePermissionsHookParams,
+  ) => Promise<null>;
+};
+
+/**
+ * The `snap_revokePermissions` method implementation.
+ * Revokes dynamic permissions at the runtime.
+ *
+ * Note: The implementation of this method should mirror
+ * the implementation of wallet_requestPermissions as close as possible.
+ * (https://github.com/MetaMask/core/blob/main/packages/permission-controller/src/rpc-methods/revokePermissions.ts#L51).
+ *
+ * @param request - The JSON-RPC request object.
+ * @param response - The JSON-RPC response object.
+ * @param _next - The `json-rpc-engine` "next" callback. Not used by this
+ * function.
+ * @param end - The `json-rpc-engine` "end" callback.
+ * @param hooks - The RPC method hooks.
+ * @param hooks.getOriginSnap - Get Snap hook.
+ * @param hooks.revokePermissions - Revoke permission hook.
+ */
+async function revokePermissionsImplementation(
+  request: JsonRpcRequest<RevokePermissionsParams>,
+  response: PendingJsonRpcResponse<RevokePermissionsResult>,
+  _next: unknown,
+  end: JsonRpcEngineEndCallback,
+  { getOriginSnap, revokePermissions }: RevokePermissionsHooks,
+): Promise<void> {
+  if (!isPlainObject(request.params)) {
+    return end(rpcErrors.invalidParams({ data: { request } }));
+  }
+
+  const snap = getOriginSnap();
+
+  const permissionsToRevoke = request.params as SnapPermissions;
+  const permissionsToRevokeKeys = Object.keys(
+    permissionsToRevoke,
+  ) as RevokePermissionsHookParams;
+
+  const initialPermissions = snap?.manifest.initialPermissions;
+  const dynamicPermissions = snap?.manifest.dynamicPermissions;
+
+  assert(initialPermissions, 'Failed to get Snap initial permissions.');
+  if (!dynamicPermissions) {
+    return end(
+      rpcErrors.invalidRequest({
+        message: 'This Snap have no dynamic permissions specified.',
+      }),
+    );
+  }
+
+  for (const permissionKey of permissionsToRevokeKeys) {
+    if (hasProperty(initialPermissions, permissionKey)) {
+      return end(
+        rpcErrors.invalidRequest({
+          message: `A permission (${String(
+            permissionKey,
+          )}) requested for revocation is not dynamic permission.`,
+        }),
+      );
+    }
+
+    if (!hasProperty(dynamicPermissions, permissionKey)) {
+      return end(
+        rpcErrors.invalidRequest({
+          message: `A permission (${String(
+            permissionKey,
+          )}) requested for revocation is not in the manifest's dynamic permission specification.`,
+        }),
+      );
+    }
+  }
+
+  response.result = await revokePermissions(permissionsToRevokeKeys);
```

Comment:
There is no result here, we should call revokePermissions and set result to `null` ourselves afterwards.